aten/src/ATen/functorch/ADInterpreters.cpp

```diff
@@ -118,6 +119,24 @@ static void autogradBasedTransformSendToNext(
   for (const auto arg_idx : c10::irange(0, args_size)) {
     stack->push_back((*stack)[front + arg_idx]);
   }
+
+  int64_t immutable_inputs = 0;  // a bitset of the immutable inputs
```

Comment:
Use `std::bitset<64>` instead of raw int64_t for type-safety; this will also make the code nicer to read (there are APIs like bitset[bit] = ... and bitset::any() that make it so we don't need to do all the bit manipulation below).